vcstools/prof_utils.py

```diff
@@ -555,6 +578,470 @@ def check_clip(prof_to_check, toomuch=0.8, toolittle_frac=0., toolittle_absolute
+            else:  # Right side
+                underestimate = [round(dy2_roots[i-1]), round(dy2_roots[0])]
+        underest_on_pulse.append(underestimate)
+        
+        # Overestimated on-pulse
+        false_max = np.append(false_max, root)
+        false_max = sorted(false_max)
+        j = false_max.index(root)
+        false_max.remove(root)
+        # Append the overestimate
+        try:
+            overestimate = [round(false_max[j-1]), round(false_max[j])]
+        except IndexError as e:
+            # The next inflection is on the other side of the profile
+            if j == 0:  # Left side
+                overestimate = [round(false_max[-1]), round(false_max[j])]
+            else:  # Right side
+                overestimate = [round(false_max[j-1]), round(false_max[0])]
+        # If the bounds are touching
+        if overest_on_pulse and overestimate[0] == overest_on_pulse[-1][1]:
+            overest_on_pulse[-1][1] = overestimate[1]
+        # If the last element wraps around to the first
+        elif overest_on_pulse and overestimate[1] == overest_on_pulse[0][0]:
+            overest_on_pulse[0][0] = overestimate[0]
+        else:
+            overest_on_pulse.append(overestimate)
+    
+    # Remove any duplicate items because somehow this happens sometimes
+    overest_on_pulse.sort()
+    overest_on_pulse = list(k for k, _ in itertools.groupby(overest_on_pulse))
+    underest_on_pulse.sort()
+    underest_on_pulse = list(
+        k for k, _ in itertools.groupby(underest_on_pulse))
+
+    # Fill the on-pulse regions
+    overest_on_pulse = _fill_on_pulse_region(
+        overest_on_pulse, splined_profile, norm_prof, clip_noise)
+    underest_on_pulse = _fill_on_pulse_region(
+        underest_on_pulse, splined_profile, norm_prof, clip_noise)
+    # Get the off-pulse pairs as well
+    overest_off_pulse = get_off_pulse(overest_on_pulse)
+    underest_off_pulse = get_off_pulse(underest_on_pulse)
+
+    # Work out the noise std and mean using the oversetimated on-pulse region
+    off_pulse_noise_std = np.nanstd(
+        profile_region_from_pairs(norm_prof, overest_off_pulse))
+    off_pulse_noise_mean = np.nanmean(
+        profile_region_from_pairs(normamlise_prof(profile), overest_off_pulse))
+
+    # Sort the maxima based on amplitude
+    maxima_amps = []
+    for m in real_max:
+        maxima_amps.append(splined_profile[m])
+    real_max = [i for _, i in sorted(zip(maxima_amps, real_max), reverse=True)]
+
+    # plot if necessary
+    if plot_name:
+        plt.figure(figsize=(14, 10))
+        plt.plot(x, norm_prof, label="Profile", color="cornflowerblue")
+        plt.plot(x, reg_prof, label="Smoothed Profile", color="orange")
+        plt.xlim((0, len(x)-1))
+        for i in real_max:
+            plt.axvline(x=i, ls=":", lw=2, color="gray")
+        for i in overest_on_pulse:
+            plt.axvline(x=i[0], ls=":", lw=2, color="red")
+            plt.axvline(x=i[1], ls=":", lw=2, color="red")
```

Comment:
Do we always use the overest_on_pulse in the future calculations? If not it may be worth labelling this so it's clear. Also attaching a plot to this pull request would be cool so I can have a quick look

Reply:
Depends on what you're calculating. The overestimate is just that - it overestimates how much of the profile is on-pulse.
![on_pulse](https://user-images.githubusercontent.com/34429506/123910909-af62f600-d9ad-11eb-9284-57c5819e5205.png)

Reply:
I'm thinking of the SN calculations. It would be nice to have a plot that showed where the on and off pulse is for that calculation, is there such a plot?

Reply:
you're looking at it